### PR TITLE
Layer optimizations

### PR DESCRIPTION
**Pull request by Garanas**
This caches the GetCurrentLayer() outcome each time when OnLayerChange() is called by the engine. In turn, we save on engine calls while at the same time not breaking compatibility with mods and / or unchanged code.

### CODE REVIEW

File: units/XSB1102/XSB1102_script.lua [hunk 1]
```diff
@@ -5,7 +5,7 @@
 --#**
 --#**  Summary  :  Seraphim Hydrocarbon Power Plant Script
 --#**
---#**  Copyright � 2007 Gas Powered Games, Inc.  All rights reserved.
+--#**  Copyright � 2007 Gas Powered Games, Inc.  All rights reserved.
```
Comment (Sheikah45) [inline units/XSB1102/XSB1102_script.lua:8]: Ha seems like there might be an unintentional encoding change

Comment (Garanas) [reply units/XSB1102/XSB1102_script.lua:8]: Ah, you are correct.

File: engine/Sim/Unit.lua [hunk 2]
```diff
@@ -130,7 +130,8 @@ end
 function Unit:GetConsumptionPerSecondMass()
 end
 
---- Return the name of the layer the unit is currently in.
+--- Return the name of the layer the unit is currently in. This value is cached inside
+-- unit.Layer each time a layer changes (when OnLayerChanged is called).
```
Comment (KionX) [inline engine/Sim/Unit.lua:134]: OnLayerChange

**Comment by KionX** [comment]
There is no inherited call OnLayerChange in defaultunits.lua 2145...

**Comment by Garanas** [comment]
I think you are referring to https://github.com/FAForever/fa/blob/deploy/fafdevelop/lua/defaultunits.lua#L2225 . You are right, missed that one.

**Comment by Garanas** [comment]
Closed in favor of https://github.com/FAForever/fa/pull/3366 - which is essentially the same but because of my git setup the previous one was on FAForever and this one is coming from a fork. Also easily fixes the encoding problem Sheikah pointed out.

**Pull request closed without merging**